src/components/manager/ManagerDropBoxContent.js

```diff
@@ -194,15 +193,10 @@ const FileDisplay = ({file, tree, treeLoading}) => {
                     </Document>
                 );
             } else if (fileExt === "json") {
-                return (
-                    <ReactJson
-                        src={JSON.parse(fileContents[file] || "{}")}
-                        displayDataTypes={false}
-                        enableClipboard={false}
-                        name={null}
-                        collapsed={true}
-                    />
-                );
+                if (Array.isArray(fileContents[file])) {
+                    return (<JsonArrayDisplay doc={fileContents[file] || []} standalone/>);
+                }
+                return (<JsonObjectDisplay doc={fileContents[file] || {}}/>);
```

Comment:
this being the default case means that the title "JSON object" shows when it is loading. I think it would be better if it showed nothing while loading.

also this will break on JSON primitives - so if the file just has a string or number in it. while this isn't hugely important for all drop box files we see, I think for completeness it would be good to check if it is 

* an array
* another form of object
* else: just render in a plain ReactJson (for primitives)

while loading, you can just render a `<div />`.

Reply:
Now returning a div when loading.
Also wrapped JsonObjectDisplay and JsonArrayDisplay inside of JsonDisplay for usability. Fixed primitive behaviour in JsonPropertyDisplay.